rules/php70/tests/Rector/FuncCall/NonVariableToVariableOnFunctionCallRector/Fixture/reset_over_static_call.php.inc

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Rector\Php70\Tests\Rector\FuncCall\NonVariableToVariableOnFunctionCallRector\Fixture;
+
+class ResetOverStaticCall {
+    public static function test() {}
+}
+
+function run() {
+    return reset(ResetOverStaticCall::test());
+}
+
```

Comment:
I don't see any `Function_` node added in a PR.
Why a function here? It makes test fixture more complex then necessary

Reply:
Otherwise, it fails at compile-time on newer PHP versions

Reply:
I see. In that case, it should be in stanadlone test case that requires this PHP version.

E.g. this test case works only with PHP 7.4, https://github.com/rectorphp/rector/blob/a15d2f8ccc93abaa0981c0daa536020af9885869/rules/restoration/tests/Rector/Property/MakeTypedPropertyNullableIfCheckedRector/MakeTypedPropertyNullableIfCheckedRectorTest.php#L15

While the rest of tests are tested on PHP 7.2-7.4 now

Reply:
Test without function-wrapping will fail at PHP 8+. Meanwhile, the Rector will work correctly and the test will pass. For me, it means, that when we abandon PHP 7 testing, this test will be skipped and the logic might be silently broken.

IMHO, wrapping in `function` is more durable than `@requires PHP < 8.0`, but I can make this test `< 8.0` if you insist.

Reply:
I want code to be parsed, not executed immediately.
There is a lot of function-wraps without any additional reason and it's perfectly explainable

http://github.com/SilverFire/rector/blob/57d2394a528c916c6603ecc2da73b5d70932f5d9/rules/php73/tests/Rector/FuncCall/RemoveMissingCompactVariableRector/Fixture/fixture.php.inc#L5-L14

http://github.com/SilverFire/rector/blob/57d2394a528c916c6603ecc2da73b5d70932f5d9/rules/php80/tests/Rector/Switch_/ChangeSwitchToMatchRector/Fixture/skip_different_variable_assign.php.inc#L5-L18

http://github.com/SilverFire/rector/blob/57d2394a528c916c6603ecc2da73b5d70932f5d9/rules/php80/tests/Rector/NotIdentical/StrContainsRector/Fixture/fixture.php.inc#L5-L12

Reply:
> but I can make this test < 8.0 if you insist.

:+1: It's the convention here, it's clear what it does and work for all the other future PHP feaures.

Reply:
Meh, for `PHP < 8.0` it's still a notice. I can't see any other options except wrapping it in a `function`

Reply:
Ok, I'll look at it locally after merge to get better understanding.